File: Set/448_Find_All_Numbers_Disappeared_in_an_Array/448_Find_All_Numbers_Disappeared_in_an_Array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        Set<Integer> numSet = new HashSet<>();
        List<Integer> res = new ArrayList<>();
        int n = nums.length;

        for (int num : nums) {
            numSet.add(num);
        }

        for (int i = 1; i <= n; i++) {
            if (!numSet.contains(i)) {
                res.add(i);
            }
        }
        return res;        
    }
}

File: Set/448_Find_All_Numbers_Disappeared_in_an_Array/448_Find_All_Numbers_Disappeared_in_an_Array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int n = nums.size();
        vector<int> ref(n);
        iota(ref.begin(), ref.end(), 1);
        sort(nums.begin(), nums.end()); 
        
        vector<int> res(n); 
        vector<int>::iterator it = set_difference(ref.begin(), ref.end(), nums.begin(), nums.end(), res.begin());
        res.resize(it-res.begin());  
        return res;
    }
};